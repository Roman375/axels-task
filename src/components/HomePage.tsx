import React, { FC } from 'react'
import { Route, Switch } from 'react-router'
import { BrowserRouter as Router } from 'react-router-dom'

import { Container } from '../styled/styled'
import { UsersList, OverviewPage } from '../components'

type Props = {
  getEmployee: () => void
  getOverviews: (employeeName: string) => void
  employees: {
    employees: string[]
  }
  overviews: Array<object>
}

const HomePage: FC<Props> = ({ getEmployee, getOverviews, ...props }) => (
  <Router>
    <Container>
      <Switch>
        <Route
          exact
          path="/"
          render={() => (
            <UsersList
              getEmployee={getEmployee}
              employees={props.employees}
              getOverviews={getOverviews}
            />
          )}
        />
        <Route
          exact
          path="/overview/:employee"
          render={() => <OverviewPage overview={props.overviews} />}
        />
      </Switch>
    </Container>
  </Router>
)

export default HomePage
