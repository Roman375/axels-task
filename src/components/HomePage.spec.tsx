import React from 'react'
import { mount } from 'enzyme'
import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import '@testing-library/jest-dom/extend-expect'

import HomePage from './HomePage'

describe('<HomePage />', () => {
  const props = {
    getEmployee: () => {},
    getOverviews: () => {},
    employees: {
      employees: ['john', 'ben'],
    },
    overviews: [],
  }

  describe('HomePage render <UsersList />', () => {
    const homePage = mount(<HomePage {...props} />)

    it('render properly', () => {
      expect(homePage).toMatchSnapshot()
    })

    expect(homePage.find('UsersList')).toHaveLength(1)
  })

  describe('full app rendering/navigating', () => {
    const renderWithRouter = (ui: any, { route = '/' } = {}) => {
      window.history.pushState({}, 'Test page', route)

      return render(ui, { wrapper: MemoryRouter })
    }

    it('Router UsersList', () => {
      renderWithRouter(<HomePage {...props} />)
      expect(screen.getByText(/john/i)).toBeInTheDocument()
    })

    it('Router OverviewPage', () => {
      renderWithRouter(<HomePage {...props} />, {
        route: '/overview/:employee',
      })

      expect(screen.getByText(/Overview/i)).toBeInTheDocument()
    })
  })
})
