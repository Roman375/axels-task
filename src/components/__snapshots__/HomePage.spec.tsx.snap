// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> HomePage render <UsersList /> render properly 1`] = `ReactWrapper {}`;
